### PR TITLE
Feat/net097 6 1/add unit schema

### PR DESCRIPTION
**Pull request by Alex-knu**
Added schema for Unit(added to unlock task "(NET097-21)[Architecture] Make a solution for activity log")

### CODE REVIEW

File: MilitaryManager.Core/Entities/UnitEntity/Unit.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        public int Id { get; set; }
+        public string Name { get; set; }
+        public string Address { get; set; }
+        public int? ParentId { get; set; }
+
+        public virtual Unit Parent { get; set; }
+
+        public virtual ICollection<Unit> SubUnits { get; set; }
+
+        public List<Soldier> Soldiers { get; set; } = new List<Soldier>();
```
Comment (vladtymo) [inline MilitaryManager.Core/Entities/UnitEntity/Unit.cs]: I think it's better to use ICollection type for navigation prop and HashSet as an instance.

File: MilitaryManager.Core/Entities/SoldierEntity/Soldier.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+{
+    [Table("Soldier", Schema = "Unit")]
+    public class Soldier : IBaseEntity
+    {
+        public int Id { get; set; }
+        public string LastName { get; set; }
+        public string FirstName { get; set; }
+        public int? UnitId { get; set; }
+        public int? RankId { get; set; }
+        public int? PatronymicId { get; set; }
```
Comment (vladtymo) [inline MilitaryManager.Core/Entities/SoldierEntity/Soldier.cs]: Change the name to more appropriate.

**Review by vladtymo — CHANGES_REQUESTED** [review]

File: MilitaryManager.Core/Entities/AttributeEntity/Attribute.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
+﻿using MilitaryManager.Core.Entities.EntityToAttributeEntity;
+using MilitaryManager.Core.Entities.ProfileEntity;
+using MilitaryManager.Core.Interfaces;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace MilitaryManager.Core.Entities.AttributeEntity
+{
+    [Table("Attribute", Schema = "Unit")]
```
Comment (vladtymo) [inline MilitaryManager.Core/Entities/AttributeEntity/Attribute.cs]: Pluralize the table names.

**Review by vladtymo — APPROVED** [review]

File: MilitaryManager.Core/DTO/Units/UnitDTO.cs [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    public class UnitDTO
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+        public string Address { get; set; }
+        public int? ParentId { get; set; }
+
+        public virtual UnitDTO Parent { get; set; }
+
+        public virtual ICollection<UnitDTO> SubUnits { get; set; }
```
Comment (YevheniiVovk) [inline MilitaryManager.Core/DTO/Units/UnitDTO.cs:14]: What difference between virtual and not virtual navigation properties 
Are we sure that we need to use virtual properties

File: MilitaryManager.Core/Interfaces/IBaseEntity.cs [hunk 5]
```diff
@@ -0,0 +1,10 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace MilitaryManager.Core.Interfaces
+{
+    public interface IBaseEntity
+    {
+    }
+}
```
Comment (YevheniiVovk) [inline MilitaryManager.Core/Interfaces/IBaseEntity.cs:10]: We should not use empty interface

File: MilitaryManager.Core/Entities/EntityEntity/Entity.cs [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+{
+    [Table("Entities", Schema = "Unit")]
+    public class Entity : IBaseEntity
+    {
+        public int Id { get; set; }
+        public string RegNum { get; set; }
+
+        public ICollection<EntityToAttribute> EntityToAttributes { get; set; } = new HashSet<EntityToAttribute>();
+        public EquipmentToSoldier EquipmentToSoldiers { get; set; } = new EquipmentToSoldier();
+    }
```
Comment (YevheniiVovk) [inline MilitaryManager.Core/Entities/EntityEntity/Entity.cs:17]: IBaseEntity is empty, what a reason to implement empty interface

**Comment by YevheniiVovk** [comment]
At total let's split to 2 PR's 
First one should contains only unit 
Second ione should contains all scheme parts

File: MilitaryManager.Core/Entities/EntityToAttributeEntity/EntityToAttribute.cs [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    [Table("EntityToAttributes", Schema = "Unit")]
+    public class EntityToAttribute : IBaseEntity
+    {
+        public int Id { get; set; }
+        public int EntityId { get; set; }
+        public int AttributeId { get; set; }
+        public string Value { get; set; }
+
+        public virtual Entity Entity { get; set; }
+        public virtual Attribute Attribute { get; set; }
```
Comment (YevheniiVovk) [inline MilitaryManager.Core/Entities/EntityToAttributeEntity/EntityToAttribute.cs:17]: Same here

File: docker-compose.override.yml [hunk 8]
```diff
@@ -37,7 +38,7 @@ services:
     environment:
       - ASPNETCORE_ENVIRONMENT=Development
       - ASPNETCORE_URLS=https://+:443;http://+:80
-      - ConnectionStrings__DefaultConnection=Server=host.docker.internal;Database=IdentityServer_AspNet_Identity;User ID=Gwa_System;Password=T&t137S5tP!+3SMJQ<r0KJ;MultipleActiveResultSets=true;Application Name=IdentityServer
+      - ConnectionStrings__DefaultConnection=Server=host.docker.internal;Database=IdentityServer_AspNet_Identity;User ID=alex;Password=P@ssw0rd1;MultipleActiveResultSets=true;Application Name=IdentityServer
```
Comment (YevheniiVovk) [inline docker-compose.override.yml:41]: DO NOT COMMIT changes in docker compose

File: MilitaryManager.Core/Entities/UnitEntity/Units.cs [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+        internal class UnitsList : Specification<Unit>
+        {
+            public UnitsList()
+            {
+                Query
+                .Include(x => x.SubUnits);
+            }
+        }
+    }
+}
```
Comment (YevheniiVovk) [inline MilitaryManager.Core/Entities/UnitEntity/Units.cs:16]: Empty line

File: MilitaryManager.Core/Entities/UnitEntity/Unit.cs [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+        public int? ParentId { get; set; }
+
+        public virtual Unit Parent { get; set; }
+
+        public virtual ICollection<Unit> SubUnits { get; set; }
+
+        public ICollection<Soldier> Soldiers { get; set; } = new HashSet<Soldier>();
+        public ICollection<EquipmentToSoldier> EquipmentToSoldiers { get; set; } = new HashSet<EquipmentToSoldier>();
+    }
+}
```
Comment (YevheniiVovk) [inline MilitaryManager.Core/Entities/UnitEntity/Unit.cs:24]: Empty line

File: MilitaryManager.Core/Entities/UnitEntity/Unit.cs [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+    public class Unit : IBaseEntity
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+        public string Address { get; set; }
+        public int? ParentId { get; set; }
+
+        public virtual Unit Parent { get; set; }
+
+        public virtual ICollection<Unit> SubUnits { get; set; }
```
Comment (YevheniiVovk) [inline MilitaryManager.Core/Entities/UnitEntity/Unit.cs:19]: What difference between virtual and not virtual navigation properties.

File: MilitaryManager.Core/Entities/SoldierEntity/Soldier.cs [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+        public int RankId { get; set; }
+        public int? ParentId { get; set; }
+        public int PositionId { get; set; }
+
+        public virtual Unit Unit { get; set; }
+        public virtual Rank Rank { get; set; }
+        public virtual Soldier Parent { get; set; }
+        public virtual Position Position { get; set; }
+
+        public virtual ICollection<Soldier> SubSoldiers { get; set; }
```
Comment (YevheniiVovk) [inline MilitaryManager.Core/Entities/SoldierEntity/Soldier.cs:28]: What difference between virtual and not virtual navigation properties.

**Pull request closed without merging**